cli/gitleaks/gitleaks.py

```diff
@@ -0,0 +1,320 @@
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import os
+import shutil
+import subprocess as sp
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+from typing import Literal
+
+import requests
+from simple_logger.logger import get_logger
+
+BUILD_ID_VAR: str = "BUILD_ID"
+ARTIFACT_DIR_VAR: str = "ARTIFACT_DIR"
+
+
+@dataclass
+class GitleaksConfig:
+    _server_url: str
+    _token_path: str
+    _output_file: str
+    _gitleaks_version = "7.6.1"
+    _artifact_dir: None | Path = None
+    _job_dir: None | Path = None
+    _patterns_file_url: None | str = None
+    _keep_job_dir: None | bool = False
+    _gitleaks_report_path: None | Path = None
+    _patterns_file_path: None | Path = None
+    _env_artifact_dir: str = ""
+    _env_build_id: str = ""
+    _job_dir_root: str | Path = "/tmp"
+    _logger: Any = None
+    _keep_patterns_file: bool = False
+
+    def __post_init__(self) -> None:
+        """
+        Builds the GitleaksConfig object. This class is used to generate a Gitleaks detect scan using the latest Redhat
+        detection rules.
+
+        The expected environment variables, directories, and secrets are gathered and checked for availability.
+
+        The latest Gitleaks detection patterns are fetched from the Redhat patterns distribution server and used to run
+        a `gitleaks detect` scan on the job directory created by the `firewatch report` cli command.
+
+        If any detections are found, a report is created and archived in the ARTIFACT_DIR.
+
+        The detection patterns file and the job directory are removed during teardown.
+        """
+
+        self._get_env_vars()
+        self._check_env()
+
+    def _get_env_vars(self) -> None:
+        """
+        Get the values for the ARTIFACT_DIR and BUILD_ID environment variables.
+        """
+        self._env_artifact_dir = os.getenv(ARTIFACT_DIR_VAR, "")
```

Comment:
There is no need to set `""` as default to `os.getenv`, default is `None` and you have `_check_env` that verify that you have valid value

Reply:
The empty strings were added to satisfy the mypy pre-commit checks, but I will see if adding more specific type definitions would be better.

Reply:
From what I'm seeing, it looks like defaulting to the empty strings is the most straightforward way to satisfy the mypy checks. We could disable type checking for this value, but I think limiting the possible types to just be `str` and leaving the type check enabled would be better.

Just let me know what you think. Thanks!